### PR TITLE
Gamification, rebases #143 and  #137, closes #129 and closes #130

### LINKED ISSUE #129: Make points available (backend)

### LINKED ISSUE #130: Make a leaderboard (backend)

### PR DESCRIPTION
**Pull request by JWGmeligMeyling**
* Rebased on master
* Using id instead of soundcloudId
* Using authentication token for /me request
* Front end issue: https://github.com/MoodCat/MoodCat.me-Frontend/pull/57

### CODE REVIEW

File: src/test/resources/bootstrap/users.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+			"id": 100,
+			"username": "Gijs",
+			"soundCloudUserId": 1,
+			"points": 666
+		},
+		{
+			"id": 101,
+			"username": "Tim",
+			"soundCloudUserId": 2,
+			"points": 420
```
Comment (JeremybellEU) [inline src/test/resources/bootstrap/users.json]: Blaze it!

**Comment by JeremybellEU** [comment]
UserModel and UserAPI are completely untested, they should be tested.

SongAPI.approachSong() should also be tested. I would rename it though, since approaching seems a bit weird. Maybe `SongAPI.setInitialVector()`?

Apart from these 2 minor remarks, this branch can be merged!

P.S. I didn't verify with frontend though, so we might want to do that too and have full prove of this working version.

**Comment by JWGmeligMeyling** [comment]
Yup but this was already the case in the existing PR's that were OK'ed and I think its time to merge now. A full pass on the frondend on someone elses machine would be nice ( @GijsWeterings ?)

**Comment by GijsWeterings** [comment]
LGTM, Merging

**Pull request merged**